kuma/wiki/views/list.py

```diff
@@ -190,30 +189,32 @@ def get_previous(revisions):
 
     # Grab revisions, but defer summary and content because they can lead to
     # attempts to cache more than memcached allows.
-    revisions = MultiQuerySet(
-        (Revision.objects.filter(pk=document.current_revision.pk)
-                         .prefetch_related('creator', 'document')
-                         .transform(get_previous)),
-        (Revision.objects.filter(document=document)
-                         .order_by('-created', '-id')
-                         .exclude(pk=document.current_revision.pk)
-                         .prefetch_related('creator', 'document')
-                         .transform(get_previous))
-    )
-
-    if not revisions.exists():
+    all_revisions = (document.revisions.defer('summary', 'content').order_by('created', 'id')
+                                                                   .reverse().transform(get_previous))
```

Comment:
Found it: PR #2546.  Continue with the suggested changes, and we'll see what the perf is like with "all revisions" and a big revision history.